src/pepr/operator/controllers/keycloak/client-sync.ts

```diff
@@ -45,6 +52,7 @@ export async function purgeSSOClients(pkg: UDSPackage, refs: string[] = []) {
     const token = Store.getItem(ref);
     const clientId = ref.replace("sso-client-", "");
     if (token) {
+      Store.removeItem(ref);
       await apiCall({ clientId }, "DELETE", token);
```

Comment:
Not sure which order of these is best, I think I would put `Store.removeItem(ref);` after `await apiCall()` and wrap both of these in a try/catch. It looks like there's no error handling at the moment and the error won't be caught.

Need to decide what behavior you want in case of Keycloak API errors. My suggestion would be to not drop the client from the store unless the DELETE succeeds. Unsure what's best, if it goes wrong someone may need to manually drop the item and I don't think it's really possible to hack the Pepr store directly. Deleting an item from the `.spec` manually is possible though, I don't think the client names are stored there though?